server/controllers/notification.js

```diff
@@ -0,0 +1,108 @@
+const Notification = require("../models/Notification");
+const asyncHandler = require("express-async-handler");
+
+const notifications = {
+  newRequest: {
+    title: "New Request",
+    message: "You recieved a new request from ",
+  },
+  message: {
+    title: "New Message",
+    message: "You received a new message from ",
+  },
+  requestResponse: {
+    title: "Response Received",
+    message: "You received a response from ",
+  },
+};
+
+// @route POST /notification/:id
+// @desc create notification
+// @access Private
+exports.createNotification = asyncHandler(async (req, res) => {
+  const sender = rep.params.sender;
+  const type = req.params.type;
+  const userId = req.params.userId;
+
+  if (!sender | type | userid) {
+    res.status(400);
+    throw new Error("Bad Request");
+  }
+
+  const newNotification = await Notification.create({
+    type,
+    title: notifications.type.title,
+    description: `${notifications.type.message}${sender}`,
+    date: new Date.now(),
+    userId,
+  });
+
+  if (!newNotification) {
+    res.status(200).json({
+      success: {
+        profile: newNotification,
+      },
+    });
+  }
+  res.status(500);
+  throw new Error("Error handling request");
+});
+
+// @route GET /notification
+// @desc Get user profile data
+// @access Private
+exports.getNewNotifications = asyncHandler(async (req, res) => {
+  const newNotifcations = await Notification.find({
+    userId: req.user.id,
+    read: false,
+  });
+  if (!newNotifcations && newNotifcations == []) {
```

Comment:
Please note that `newNotifcations == []` will always be false as objects (including functions) and arrays are compared by reference in JavaScript. If you want to test for emptiness just use the `length` property. Also, if `newNotifcations` is falsy, no need for the second statement since an empty array is not falsy. So I'd suggest you use `if (newNotifcations?.length)`

Reply:
Updated the condition. I was trying to allow empty lists to pass as non faulty. Thanks for the insight.